From: Dermot Bradley <dermot_bradley@yahoo.com>
Date: Fri, 30 Oct 2020 00:46 +0000
Subject: [PATCH] openrc: remove warning from osclock

The osclock init.d script can be used instead of the hwclock init.d
script on machines with a RTC (where the driver is compiled into the
kernel) and where the kernel is configured (CONFIG_RTC_HCTOSYS=Y) to
sync the system clock during boot *and* from system clock to RTC
automatically every 10 minutes (CONFIG_RTC_SYSTOHC=Y) when the system
clock has NTP sync.

This patch adds a stub start() function to remove OpenRC warnings
that appear during boot. There is no need for the start() function
to actually do anything as the osclock script's sole purpose is simply
to meet the "clock" dependancy of other init.d scripts.

This patch was submitted upstream as PR 377.

---

diff -aur a/init.d/osclock.in b/init.d/osclock.in
--- a/init.d/osclock.in
+++ b/init.d/osclock.in
@@ -17,3 +17,15 @@
 {
 	provide clock
 }
+
+start()
+{
+	# This stub function is required to avoid OpenRC warning at boot:
+	#
+	#  * The command variable is undefined.
+	#  * There is nothing for osclock to start.
+	#  * If this is what you intend, please write a start function.
+	#  * This will become a failure in a future release.
+	#
+	return 0
+}
